.github/workflows/ci-experimental.yml

```diff
@@ -106,4 +106,9 @@ jobs:
           key: ${{ runner.os }}-vendor-modules-${{ steps.submodules.outputs.hash }}
 
       - name: Run tests
-        run: make V=1 LOG_LEVEL=DEBUG QUICK_AND_DIRTY_COMPILER=1 test2 testwakunode2
+        run: |
+          if [ ${{ runner.os }} == "Linux" ]; then
+            sudo docker run --rm -d -e POSTGRES_PASSWORD=test123 -p 5432:5432 postgres:9.6-alpine
```

Comment:
This should wait for the image pull, so slow connection should not be an issue, but in theory it could take some small amount of time to start the postgres itself, but since the postgres test is not the first to run, we should be fine. Just noting this in case we get into some "race condition" errors in the future:)

Another thing to consider is the Docker Hub rate limits - since you docker client is not authenticated, we could get rate limited on something like IP-range in case many people pull images from Docker Hub in the same time window (adding a docker login with token in secret should solve this in case it happens)

Reply:
Thanks for the comments @vpavlin !
Do you want me to tackle the auth issue now or can I merge and take this point into consideration if the problem arises?